### PR TITLE
Upgrade to Python 3.10 compatibility

### PR DESCRIPTION
**Pull request by DanielRyanSmith**
Requires updates from #33705 to function.

### CODE REVIEW

**Comment by DanielRyanSmith** [comment]
Dependency changes are merged. This change will need to be admin merged due to the "tests failing without changes" failure not taking into account the new code.

File: infrastructure/webdriver/tests/test_load_file.py [hunk 1]
```diff
@@ -1,2 +1,2 @@
-def test_load(session, inline):
+def test_load(inline):
```
Comment (foolip) [inline infrastructure/webdriver/tests/test_load_file.py:1]: Can you describe this change in the PR description? Was it necessary to pass CI, and what was going wrong here?

File: tools/ci/azure/install_python.yml [hunk 2]
```diff
@@ -1,14 +1,14 @@
...
 - powershell: |
-    $pythonPath = 'C:\Python39'
+    $pythonPath = 'C:\Python310'
     $pythonInstallerLogPath = 'python_installer.log'
     $installArgs = "/passive /log $pythonInstallerLogPath TargetDir=$pythonPath"
 
     Write-Host "Downloading python installer exe"
     $downloadPath = 'python_installer.exe'
-    Start-BitsTransfer -Source https://www.python.org/ftp/python/3.9.4/python-3.9.4-amd64.exe  -Destination $downloadPath
+    Start-BitsTransfer -Source https://www.python.org/ftp/python/3.10.2/python-3.10.2-amd64.exe  -Destination $downloadPath
```
Comment (foolip) [inline tools/ci/azure/install_python.yml:9]: Can you upgrade to 3.10.4, so that we have the most recent possible?

Comment (DanielRyanSmith) [reply tools/ci/azure/install_python.yml]: This file has been removed as of #33755.

File: infrastructure/webdriver/tests/test_load_file.py (hunk 1 above)
Comment (DanielRyanSmith) [reply infrastructure/webdriver/tests/test_load_file.py]: I believe this was a remnant of trying to solve some of the issues I was seeing with the CI. I remember feeling confident that this was a correct change that was not fixed based on how this test is called (the session argument is no longer passed to this function), but I did not document this and it's been long enough that the exact justification for it is slipping my mind. I've removed it and it seems to not affect the CI, so let's just forget about this change for now 🙂

**Comment by DanielRyanSmith** [comment]
This PR is now updated with the recent Edge changes and should again be ready for submission 🙂

**Review by foolip — APPROVED** [review]

**Pull request merged**